src/sdk/map_client.js

```diff
@@ -557,6 +563,53 @@ class MapClient {
             }));
     }
 
+    async move_blocks_to_storage_class() {
+        // Block movement is not done if move_to_tier was not requested or if insufficient
+        // previous tier information is provided.
+        if (!this.move_to_tier || !this.current_tiers || this.current_tiers?.length === 0) return;
+        if (this.current_tiers.length !== this.chunks.length) throw new Error('current_tiers length does not match chunks length');
+
+        const blocks = [];
+        const target_storage_class = this.move_to_tier.storage_class;
+        const target_attached_pools = this.move_to_tier.mirrors.map(mirror => mirror.spread_pools.map(pool => String(pool._id))).flat();
+
+        for (const [idx, chunk] of this.chunks.entries()) {
+            const tier_before_move = this.current_tiers[idx];
+            const current_attached_pools = tier_before_move.mirrors.map(mirror => mirror.spread_pools.map(pool => String(pool._id))).flat();
+
+            if (
+                !js_utils.compare_unordered(target_attached_pools, current_attached_pools, true) ||
```

Comment:
how about using _.xor() instead of creating a special function for this esoteric case?

Reply:
I .. did not know about `xor`